### PR TITLE
enhance: use lazy initializing client for streaming node

### PR DESCRIPTION
**Pull request by chyezh**
issue: #38399

### CODE REVIEW

File: internal/streamingnode/server/flusher/flusherimpl/flusher_impl.go [hunk 1]
```diff
@@ -126,11 +125,19 @@ func (f *flusherImpl) notify() {
...
-		defer f.stopWg.Done()
+		defer f.notifier.Finish(struct{}{})
+		dc, err := resource.Resource().DataCoordClient().GetWithContext(f.notifier.Context())
+		if err != nil {
+			return
+		}
+		broker := broker.NewCoordBroker(dc, paramtable.GetNodeID())
+		cpUpdater := util.NewChannelCheckpointUpdater(broker)
+		go cpUpdater.Start()
+		// When the flusher exits, the cpUpdater should be closed.
```
Comment (bigsheeper) [inline internal/streamingnode/server/flusher/flusherimpl/flusher_impl.go:138]: Could this result in being unable to update the checkpoint when the flowgraphs are closing?

File: internal/distributed/streamingnode/service.go [hunk 2]
```diff
@@ -293,34 +302,46 @@ func (s *Server) initMeta() error {
...
-	return nil
+			log.Info("StreamingNode try to wait for DataCoord ready")
+			err = componentutil.WaitForComponentHealthy(ctx, dataCoord, "DataCoord", 1000000, time.Millisecond*200)
+			if err != nil {
+				return errors.Wrap(err, "StreamingNode wait for DataCoord ready failed")
+			}
+			log.Info("StreamingNode wait for DataCoord ready")
+			s.dataCoord.Set(dataCoord)
+			return nil
+		})
```
Comment (bigsheeper) [inline internal/distributed/streamingnode/service.go:343]: retry.AttemptAlways() ?

File: internal/distributed/streamingnode/service.go [hunk 3]
```diff
@@ -293,34 +302,46 @@ func (s *Server) initMeta() error {
...
-	return nil
+			log.Info("StreamingNode try to wait for RootCoord ready")
+			err = componentutil.WaitForComponentHealthy(ctx, rootCoord, "RootCoord", 1000000, time.Millisecond*200)
+			if err != nil {
+				return errors.Wrap(err, "StreamingNode wait for RootCoord ready failed")
+			}
+			log.Info("StreamingNode wait for RootCoord done")
+			s.rootCoord.Set(rootCoord)
+			return nil
+		}, retry.AttemptAlways())
```
Comment (bigsheeper) [inline internal/distributed/streamingnode/service.go:322]: Can we throw an error after the context ends instead of retrying indefinitely?

Comment (chyezh) [reply internal/distributed/streamingnode/service.go:322]: retry will stop when error is context timeout or canceled

File: internal/streamingnode/server/flusher/flusherimpl/flusher_impl.go (hunk 1 above)
Comment (chyezh) [reply internal/streamingnode/server/flusher/flusherimpl/flusher_impl.go:138]: Got it

**Comment by chyezh** [comment]
rerun go-sdk

**Pull request merged**